.github/workflows/platform-pull-request.yml

```diff
@@ -60,23 +60,23 @@ jobs:
     name: Build Unified Image
     uses: ./.github/workflows/.reusable-docker-build.yml
     with:
-      file: Dockerfile
+      target: oss-unified
       image-name: flagsmith
```

Comment:
Shouldn't these be tagged with the PR somewhere so they don't pollute the main release images? Maybe that's happening somewhere else.

Reply:
See [here](https://github.com/Flagsmith/flagsmith/blob/ec01d411b68a23aa61f60f33275dad6d95a93632/.github/workflows/.reusable-docker-build.yml#L69-L80). The tags generated are aware of the calling GHA event: pr-prefixed for PR events, version tag and `latest` for releases, `main` for the push to main event.

Reply:
Sorry, I completely trust that what you're saying is correct I'm just trying to understand how it works a little bit (but this shouldn't hold up the PR review so if everything else is resolved, I will approve). I can see (via the link you shared) that we define the following tags: 

```
            type=ref,event=branch
            type=ref,event=tag
            type=ref,event=pr
```

... but what I'm confused about I guess is why we define all 3 there. What determines which of the three it is, and what determines the actual tag name?

Reply:
The tags input is detailed [in the action docs](https://github.com/docker/metadata-action/tree/v5/?tab=readme-ov-file#tags-input).

According to the rules we define, the built images will be tagged:
- On PR events: `pr-` prefixed.
- On merge to main: `main`.
- On tag: tag name, semver full version, semver major.minor version, `latest`.

As we publish to Docker Hub, the images are re-tagged:
- On tag: tag name, semver full version, semver major.minor version, `latest`.